numpy/lib/histograms.py

```diff
@@ -348,13 +348,15 @@ def _unsigned_subtract(a, b):
     }
     dt = np.result_type(a, b)
     try:
-        dt = signed_to_unsigned[dt.type]
+        unsigned_dt = signed_to_unsigned[dt.type]
     except KeyError:
         return np.subtract(a, b, dtype=dt)
     else:
         # we know the inputs are integers, and we are deliberately casting
-        # signed to unsigned
-        return np.subtract(a, b, casting='unsafe', dtype=dt)
+        # signed to unsigned.  The input may be negative python integers so
+        # ensure we pass in arrays with the initial dtype (related to NEP 50).
+        return np.subtract(np.asarray(a, dtype=dt), np.asarray(b, dtype=dt),
```

Comment:
Isn't it easier to just do `np.asanyarray(a - b, dtype=unsigned_dt)` - I don't think we're getting any speed-up doing it another way.

Reply:
It doesn't work because if `a` or `b` is a negative Python integer, the conversion will raise an error (rather being unsafe and "well defined").